sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/HyperLogLogPlusPlusHelper.scala

```diff
@@ -93,6 +94,8 @@ class HyperLogLogPlusPlusHelper(relativeSD: Double) extends Serializable {
     val value = dataType match {
       case FloatType => FLOAT_NORMALIZER.apply(_value)
       case DoubleType => DOUBLE_NORMALIZER.apply(_value)
+      case s: StringType if _value.isInstanceOf[UTF8String] =>
+        CollationFactory.getCollationKeyBytes(_value.asInstanceOf[UTF8String], s.collationId)
```

Comment:
```suggestion
      case st: StringType if _value.isInstanceOf[UTF8String] =>
        CollationFactory.getCollationKeyBytes(_value.asInstanceOf[UTF8String], st.collationId)
```

Reply:
also, let's rethink what we're doing here so that we avoid regression for `UTF8_BINARY`

if this piece of code should handle only non-binary collations, then let's do:
```
      case st: StringType if st.supportsBinaryEquality and _value.isInstanceOf[UTF8String] =>
        CollationFactory.getCollationKeyBytes(_value.asInstanceOf[UTF8String], st.collationId)
```
so that the effective code path for `UTF8_BINARY` strings remains unchanged